### PR TITLE
save a reference to created tasks, to avoid tasks disappearing

### PR DESCRIPTION
**Pull request by anio**
This change saves references for created tasks to avoid tasks disappearing. 

> Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks. A task that isn’t referenced elsewhere may get garbage collected at any time, even before it’s done.
>
> [python docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task)

Please review the link: https://textual.textualize.io/blog/2023/02/11/the-heisenbug-lurking-in-your-async-code/

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/app/components/serve/auto_scaler.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ async def consumer(self):
             if batch and (is_batch_ready or is_batch_timeout):
                 self._server_status[server_url] = False
                 # find server with capacity
-                asyncio.create_task(self.send_batch(batch, server_url))
+                task = asyncio.create_task(self.send_batch(batch, server_url))
```
Comment (awaelchli) [inline src/lightning/app/components/serve/auto_scaler.py]: You can add the comment
```
# noqa: F841
```
behind this line so flake8 will ignore the unused assignment

Comment (awaelchli) [reply src/lightning/app/components/serve/auto_scaler.py]: In addition, a brief comment for why this is needed (e.g. as you've done in the PR description with link to python docs) would be useful too :)

Comment (anio) [reply src/lightning/app/components/serve/auto_scaler.py]: Updated :)

File: src/lightning/app/components/serve/auto_scaler.py [hunk 2]
```diff
@@ -257,7 +257,9 @@ async def consumer(self):
             if batch and (is_batch_ready or is_batch_timeout):
                 self._server_status[server_url] = False
                 # find server with capacity
-                asyncio.create_task(self.send_batch(batch, server_url))
+                # Saving a reference to the result of this function, protects the task disappearing mid-execution
+                # https://textual.textualize.io/blog/2023/02/11/the-heisenbug-lurking-in-your-async-code/
```
Comment (awaelchli) [inline src/lightning/app/components/serve/auto_scaler.py:261]: Let's replace it with the Python docs instead, reducing the risk that the link gets outdated
```suggestion
                # https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task
```

**Review by awaelchli — APPROVED** [review]

File: src/lightning/app/components/serve/auto_scaler.py (hunk 2 above)
Comment (anio) [reply src/lightning/app/components/serve/auto_scaler.py]: Updated :)

**Review by lantiga — APPROVED** [review]
Thank you @anio, sneaky one

**Pull request merged**